Can't compile models on a M3 Macbook Pro

This is on an M3 Mac

**To Reproduce**
``` r
library("primarycensored")
pcd_model <- pcd_cmdstan_model()
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:858:47: error: non-constant-expression cannot be narrowed from type 'size_t' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
#>   858 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
#>       |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:944:18: note: in instantiation of function template specialization 'pcens_model_model_namespace::primarycensored_cdf<int, std::vector<double>, int, std::vector<double>, nullptr>' requested here
#>   944 |                  primarycensored_cdf(d, dist_id, params, pwindow,
#>       |                  ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:1028:21: note: in instantiation of function template specialization 'pcens_model_model_namespace::primarycensored_lcdf<int, std::vector<double>, int, std::vector<double>, nullptr>' requested here
#>  1028 |     log_cdf_upper = primarycensored_lcdf(d_upper, dist_id, params, pwindow,
#>       |                     ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:2813:15: note: in instantiation of function template specialization 'pcens_model_model_namespace::primarycensored_lpmf<false, int, std::vector<double>, int, std::vector<double>, nullptr>' requested here
#>  2813 |               primarycensored_lpmf<false>(
#>       |               ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:3257:12: note: in instantiation of function template specialization 'pcens_model_model_namespace::pcens_model_model::log_prob_impl<false, false, Eigen::Matrix<double, -1, 1>, Eigen::Matrix<int, -1, 1>, nullptr, nullptr, nullptr>' requested here
#>  3257 |     return log_prob_impl<propto__, jacobian__>(params_r, params_i, pstream);
#>       |            ^
#> /Users/eidesfun/.cmdstan/cmdstan-2.35.0/stan/src/stan/model/model_base_crtp.hpp:93:50: note: in instantiation of function template specialization 'pcens_model_model_namespace::pcens_model_model::log_prob<false, false, double>' requested here
#>    93 |     return static_cas
#> t<const M*>(this)->template log_prob<false, false, double>(
#>       |                                                  ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:2407:3: note: in instantiation of member function 'stan::model::model_base_crtp<pcens_model_model_namespace::pcens_model_model>::log_prob' requested here
#>  2407 |   ~pcens_model_model() {}
#>       |   ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:858:47: note: insert an explicit cast to silence this issue
#>   858 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
#>       |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
#>       |                                               static_cast<int>(       )
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:859:11: error: non-constant-expression cannot be narrowed from type 'size_t' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
#>   859 |           stan::math::size(primary_params)}, "assigning variable ids");
#>       |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:859:11: note: insert an explicit cast to silence this issue
#>   859 |           stan::math::size(primary_params)}, "assigning variable ids");
#>       |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#>       |           static_cast<int>(               )
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:858:47: error: non-constant-expression cannot be narrowed from type 'size_t' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
#>   858 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
#>       |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:944:18: note: in instantiation of function template specialization 'pcens_model_model_namespace::primarycensored_cdf<int, std::vector<stan::math::var_value<double>>, int, std::vector<stan::math::var_value<double>>, nullptr>' requested here
#>   944 |                  primarycensored_cdf(d, dist_id, params, pwindow,
#>       |                  ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:1028:21: note: in instantiation of function template specialization 'pcens_model_model_namespace::primarycensored_lcdf<int, std::vector<stan::math::var_value<double>>, int, std::vector<stan::math::var_value<double>>, nullptr>' requested here
#>  1028 |     log_cdf_upper = primarycensored_lcdf(d_upper, dist_id, params, pwindow,
#>       |                     ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:2928:15: note: in instantiation of function template specialization 'pcens_model_model_namespace::primarycensored_lpmf<false, int, std::vector<stan::math::var_value<double>>, int, std::vector<stan::math::var_value<double>>, nullptr>' requested here
#>  2928 |               primarycensored_lpmf<false>(
#>       |               ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:3257:12: note: in instantiation of function template specialization 'pcens_model_model_namespace::pcens_model_model::log_prob_impl<false, false, Eigen::Matrix<stan::math::var_value<double>, -1, 1>, Eigen::Matrix<int, -1, 1>, nullptr, nullptr, nullptr>'
#>  requested here
#>  3257 |     return log_prob_impl<propto__, jacobian__>(params_r, par
#> ams_i, pstream);
#>       |            ^
#> /Users/eidesfun/.cmdstan/cmdstan-2.35.0/stan/src/stan/model/model_base_crtp.hpp:98:50: note: in instantiation of function template specialization 'pcens_model_model_namespace::pcens_model_model::log_prob<false, false, stan::math::var_value<double>>' requested here
#>    98 |     return static_cast<const M*>(this)->template log_prob<false, false>(theta,
#>       |                                                  ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:2407:3: note: in instantiation of member function 'stan::model::model_base_crtp<pcens_model_model_namespace::pcens_model_model>::log_prob' requested here
#>  2407 |   ~pcens_model_model() {}
#>       |   ^
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:858:47: note: insert an explicit cast to silence this issue
#>   858 |         std::vector<int>{dist_id, primary_id, stan::math::size(params),
#>       |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
#>       |                                               static_cast<int>(       )
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:859:11: error: non-constant-expression cannot be narrowed from type 'size_t' (aka 'unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
#>   859 |           stan::math::size(primary_params)}, "assigning variable ids");
#>       |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#> /var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.hpp:859:11: note: insert an explicit cast to silence this issue
#>   859 |           stan::math::size(primary_params)}, "assigning variable ids");
#>       |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#>       |           static_cast<int>(               )
#> 4 errors generated.
#> make: *** [/var/folders/n9/h_419gjj2mg3d208nplgvbg40000gp/T/RtmpBrnglH/model-11fed4638ba62.o] Error 1
#> Error: An error occured during compilation! See the message above for more information.
```

<sup>Created on 2024-12-10 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

--- a/NEWS.md
+++ b/NEWS.md
@@ -11,6 +11,7 @@ Development release.
 ## Bug fixes
 
 - Added a missing `@family` tag to the `pcens` functions. This omission resulted in the Weibull analytical solution not being visible in the package documentation.
+- Changed a call to `size()` to use `num_elements()` instead as an underlying type conversion was causing issues on some platforms.
 
 # primarycensored 1.0.0
 

--- a/inst/stan/functions/primarycensored.stan
+++ b/inst/stan/functions/primarycensored.stan
@@ -37,8 +37,8 @@ real primarycensored_cdf(data real d, int dist_id, array[] real params,
   } else {
     // Use numerical integration for other cases
     real lower_bound = max({d - pwindow, 1e-6});
-    array[size(params) + size(primary_params)] real theta = append_array(params, primary_params);
-    array[4] int ids = {dist_id, primary_id, size(params), size(primary_params)};
+    array[num_elements(params) + num_elements(primary_params)] real theta = append_array(params, primary_params);
+    array[4] int ids = {dist_id, primary_id, num_elements(params), num_elements(primary_params)};
 
     vector[1] y0 = rep_vector(0.0, 1);
     result = ode_rk45(primarycensored_ode, y0, lower_bound, {d}, theta, {d, pwindow}, ids)[1, 1];

